packages/contract/lib/contract/constructorMethods.js

```diff
@@ -128,14 +150,12 @@ module.exports = Contract => ({
   },
 
   setNetworkType(networkType = "ethereum") {
-    if (this.web3) {
-      this.web3.setNetworkType(networkType);
-    }
-
-    this.networkType = networkType;
+    this.configureNetwork({ networkType });
   },
 
   setWallet(wallet) {
```

Comment:
~Do we really even support this method?~

[Apparently we do, just not with `geth`](https://github.com/trufflesuite/truffle/blob/develop/packages/contract/test/methods.js#L539-L541).